xarray/backends/zarr.py

```diff
@@ -816,11 +816,29 @@ def open_store_variable(self, name):
             if zarr_array.fill_value is not None:
                 attributes["_FillValue"] = zarr_array.fill_value
         elif "_FillValue" in attributes:
-            attributes["_FillValue"] = FillValueCoder.decode(
-                attributes["_FillValue"], zarr_array.dtype
-            )
+            # TODO update version check for the released version with dtypes
+            #  probably be 3.1
+
+            # if Version(zarr.__version__) > Version("3.0.10"):
+            if hasattr(zarr_array.metadata.data_type, "to_native_dtype"):
+                native_dtype = zarr_array.metadata.data_type.to_native_dtype()
+                attributes["_FillValue"] = (
+                    # Use the new dtype infrastructure instead of doing xarray
+                    # specific fill value decoding
+                    FillValueCoder.decode(
+                        attributes["_FillValue"],
+                        native_dtype,
+                    )
+                )
+            else:
+                dtype_value = zarr_array.metadata.data_type.value
```

Comment:
same here -- I don't think you need to go through the metadata to get the numpy dtype

Reply:
also I'm confused because I thought I [fixed this already](https://github.com/pydata/xarray/pull/10348/files)

Reply:
I've had this since my initial test, never thought to try to remove it